generation/new_client_hermetic_build/new-client.py

```diff
@@ -0,0 +1,297 @@
+
+    if api_shortname == "":
+        sys.exit("api_shortname is empty")
+
+    path_to_yaml = os.path.join(script_dir, "..", "..", "generation_config.yaml")
+    with open(path_to_yaml, "r") as file_stream:
+        config = yaml.load(file_stream)
+
+    version_matcher = re.compile(r'v\d[\w\d]*')
+    proto_paths = __get_proto_paths(proto_path, config['googleapis_commitish'])
+
+    new_library = {
+        "api_shortname": api_shortname,
+        "name_pretty": name_pretty,
+        "product_documentation": product_docs,
+        "api_description": api_description,
+        "client_documentation": client_documentation,
+        "release_level": release_level,
+        "distribution_name": distribution_name,
+        "library_name": library_name,
+        "api_id": api_id,
+        "library_type": library_type,
+        "group_id": group_id,
+        "cloud_api": cloud_api,
+        "GAPICs": [{"proto_path": p} for p in proto_paths],
+    }
+
+    __add_item_if_set(new_library, "requires_billing", requires_billing)
+    __add_item_if_set(new_library, "rest_documentation", rest_docs)
+    __add_item_if_set(new_library, "rpc_documentation", rpc_docs)
+    __add_item_if_set(new_library, "distribution_name", distribution_name)
+    __add_item_if_set(new_library, "api_reference", api_reference)
+    __add_item_if_set(new_library, "codeowner_team", codeowner_team)
+    __add_item_if_set(new_library, "excluded_dependencies", excluded_dependencies)
+    __add_item_if_set(new_library, "excluded_poms", excluded_poms)
+    __add_item_if_set(new_library, "googleapis_commitish", googleapis_committish)
+    __add_item_if_set(new_library, "issue_tracker", issue_tracker)
+    __add_item_if_set(new_library, "extra_versioned_modules", extra_versioned_modules)
+
+    config["libraries"].append(new_library)
+    config["libraries"] = sorted(config["libraries"], key=__compute_library_name)
+
+    with open(path_to_yaml, "w") as file_stream:
+        yaml.dump(config, file_stream)
+
+
+def __add_item_if_set(target: dict, key: str, value: any) -> None:
+    if value is not None:
+        target[key] = value
+
+
+def __compute_library_name(library: dict) -> str:
+    if "library_name" in library:
+        return f'java-{library["library_name"]}'
+    return f'java-{library["api_shortname"]}'
+
+def __get_proto_paths(proto_path: str, committish: str) -> List[str]:
+    version_re = re.compile(r'v\d[\w\d]*')
+    is_library_version = lambda p: version_re.match(p.split('/')[-1]) is not None
+
+    if is_library_version(proto_path):
+      return [proto_path]
+
+    # if the proto path is not versioned, we must search for all versions
+    # in googleapis
+    if os.path.isdir('./googleapis'):
+      googleapis_repo = Repo('./googleapis')
+    else:
+      googleapis_repo = Repo.clone_from("https://github.com/googleapis/googleapis", './googleapis')
+    googleapis_repo.git.checkout(committish)
+    subdirs = os.walk(f'./googleapis/{proto_path}')
```

Comment:
Which proto_path versions should be published?

 - We check if `BUILD` file exists at each proto_path 
 - we check the existence gradle_assembly package rule

Reply:
We still want to do a verbatim copy if the version was passed directly

Reply:
Done